app/controllers/catalog_controller.rb

```diff
@@ -358,6 +358,32 @@ def solrdoc
     render xml: xml.add( @obj.to_solr )
   end
 
+  # displays values and pagination links for a single facet field
+  def facet
+    @pagination = get_facet_pagination(params[:id], params)
+    total_pages_count(@pagination, params)
+
+    respond_to do |format|
+      # Draw the facet selector for users who have javascript disabled:
+      format.html
+      format.json { render json: render_facet_list_as_json }
+
+      # Draw the partial for the "more" facet modal window:
+      format.js { render layout: false }
+    end
+  end
+
+  def total_pages_count(pagination, params)
+    tmp_pagination = pagination
+    count = tmp_pagination.current_page || 1
+    while tmp_pagination.has_next?
```

Comment:
@mcritchlow Yep, I think it will be nice to calculate the `page_count` with just one query. @VivinChu Is it possible to calculate `page_count` through `@pagination#total_count` (total count of all_facet_values) or not?

Reply:
@lsitu - I tried the @pagination#total_count (all_facet_values.size) before and it didn't return the count that I wanted.  It returns the count of facet values for current page, not for all pages. For example if there are 22 facet values, the total_count for first page is 21 and second page is 2.  I think I'll try the example query that Matt mentioned above.   Thanks

Reply:
@mcritchlow - I tried your example query and the numFound-count is the same
 for facet.prefix=A, facet.prefix=B or for any other letter. For example http://lib-hydratail-prod.ucsd.edu:8983/solr/blacklight/select?q=*&facet=true&facet.prefix=B&facet.limit=-1&facet.field=subject_topic_sim&wt=json&rows=0
http://lib-hydratail-prod.ucsd.edu:8983/solr/blacklight/select?q=*&facet=true&facet.prefix=B&facet.limit=-1&facet.field=subject_topic_sim&wt=json&rows=0
http://lib-hydratail-prod.ucsd.edu:8983/solr/blacklight/select?q=*&facet=true&facet.prefix=C&facet.limit=-1&facet.field=subject_topic_sim&wt=json&rows=0
Do you have any idea why that happened?  Thanks

Reply:
@VivianChu - ah, yes. Numfound will not be super useful in this context. if you look at the `jq` snippet where i get the count, it's keying off the `facet_counts.facet_fields.subject_topic_sim` value (and then doing some filtering and a length operation after that to get the count).

as i said originally, there may be a 'better' way to do this. but that's the way i got the count i referenced in my example. I couldn't find a way to get a  nice number like we get with numFound, but i also didn't spent much time on it. maybe there is a way!

Reply:
@mcritchlow - The code has been updated, deployed to QA and reviewed by Gabriela.  Could you review again?  Thanks